### PR TITLE
[runtime] Use MonoError for mono_string_new_utf16

### PR DESCRIPTION
**Pull request by lambdageek**

### CODE REVIEW

**Comment by lambdageek** [comment]
build

File: mono/metadata/marshal.c [hunk 1]
```diff
@@ -10258,20 +10264,30 @@ ves_icall_System_Runtime_InteropServices_Marshal_PtrToStringUni (guint16 *ptr)
...
-		mono_set_pending_exception (mono_get_exception_argument_null ("ptr"));
-		return NULL;
+		res = NULL;
+		mono_error_set_argument_null (&error, "ptr", "");
 	} else {
-		return mono_string_new_utf16 (domain, ptr, len);
+		res = mono_string_new_utf16_checked (domain, ptr, len, &error);
 	}
+
+	if (!mono_error_ok (&error))
```
Comment (kumpera) [inline mono/metadata/marshal.c]: We should convert this to use mono_error_set_pending_exception.

File: mono/metadata/object.c [hunk 2]
```diff
@@ -5271,20 +5293,24 @@ mono_string_new_len (MonoDomain *domain, const char *text, guint length)
 {
 	MONO_REQ_GC_UNSAFE_MODE;
 
+	MonoError merror;
```
Comment (kumpera) [inline mono/metadata/object.c]: Typo, >m<error.

File: mono/metadata/object.c [hunk 3]
```diff
@@ -5299,22 +5325,27 @@ mono_string_new (MonoDomain *domain, const char *text)
...
```
Comment (kumpera) [inline mono/metadata/object.c]: Same error.

File: mono/metadata/object.c (hunk 2 above)
Comment (kumpera) [inline mono/metadata/object.c]: I see the issue, there's a conflicting "GError *error" here.

We should rename the GError as we'll eventually add a bunch of macros around having a var named error around
and this functions will not work with it.

**Comment by kumpera** [comment]
LGTM, has some trivial issues.

**Comment by lambdageek** [comment]
@monojenkins merge

**Comment by lambdageek** [comment]
build

**Pull request merged**